### PR TITLE
resolve: Add package with +cuda+klu support

### PR DESCRIPTION
**Pull request by cameronrutherford**
This is currently a private repo, but we are close to approval and so we want a public spack package to make development easier. This build has been tested to work on machines at PNNL, and will be tested on ORNL platforms in the near future. We will work on adding `spack test` support.

I also remembered that I changed my GitHub username to be all lowercase recently, so I have updated that in the relevant packages that I currently maintain.

cc @pelesh and @ryandanehy as I have made you maintainers. I probably need @pelesh (or @kswirydo)

### CODE REVIEW

File: var/spack/repos/builtin/packages/resolve/package.py
Comment (cameronrutherford) [inline var/spack/repos/builtin/packages/resolve/package.py]: @pelesh this is causing spack's CI to fail :)

Comment (pelesh) [inline var/spack/repos/builtin/packages/resolve/package.py]: This is my suggestion:
```suggestion
    """ReSolve is a library of GPU-resident sparse linear solvers. It contains iterative and direct solvers designed to run on NVIDIA and AMD GPUs, as well as CPU devices."""
```
@kswirydo, please chime in.

File: var/spack/repos/builtin/packages/resolve/package.py
Comment (pelesh) [inline var/spack/repos/builtin/packages/resolve/package.py]: Some SuiteSparse packages do have CUDA backends, but we don't use them. I sligtly eddited this comment to reflect that:
```suggestion
    # # We do not use CUDA modules of suite-sparse so we don't specify cuda_arch
    # # For other dependencies we could
```

File: var/spack/repos/builtin/packages/resolve/package.py
Comment (pelesh) [inline var/spack/repos/builtin/packages/resolve/package.py]: I suggest we have the first release versioned 1.0. @kswirydo, what do you think?
```suggestion
    # version("1.0.0", submodules=False, branch=develop)
    # version("develop", submodules=False, branch=develop)
```

File: var/spack/repos/builtin/packages/resolve/package.py
Comment (pelesh) [inline var/spack/repos/builtin/packages/resolve/package.py:45]: I think we could add CUDA arches 70,75,80 as default. This is where the code is tested. Or we can request fat CUDA binary with arch 70 and all other up.

**Review by pelesh — COMMENTED** [review]
LGTM. I added some suggestions to consider.

File: var/spack/repos/builtin/packages/resolve/package.py
Comment (cameronrutherford) [inline var/spack/repos/builtin/packages/resolve/package.py:45]:
```suggestion
        if "+cuda" in spec:
            cuda_arch_list = spec.variants["cuda_arch"].value
            if cuda_arch_list[0] != "none":
                args.append(self.define("CMAKE_CUDA_ARCHITECTURES", cuda_arch_list))
            else:
                args.append(self.define("CMAKE_CUDA_ARCHITECTURES", "70;75;80"))
```

Comment (cameronrutherford) [reply var/spack/repos/builtin/packages/resolve/package.py:45]: I don't have a reference impl for this but I added something that will attempt to use default values.

File: var/spack/repos/builtin/packages/resolve/package.py
Comment (pelesh) [inline var/spack/repos/builtin/packages/resolve/package.py]:
```suggestion
    homepage = "https://github.com/ORNL/ReSolve"
    url = "https://github.com/ORNL/ReSolve"
```
I would just use URL where Re::Solve will be published in anticipation this will happen soon. :)

Comment (cameronrutherford) [reply var/spack/repos/builtin/packages/resolve/package.py]: Perhaps we can just leave this as a draft until we have a public URL? If we want to merge, I would rather have a functioning, private repo than have to edit the link to use the package.

Comment (pelesh) [reply var/spack/repos/builtin/packages/resolve/package.py]: Agreed! We can change this once we move to GitHub.

File: var/spack/repos/builtin/packages/resolve/package.py
Comment (pelesh) [inline var/spack/repos/builtin/packages/resolve/package.py]:
```suggestion
    maintainers("cameronrutherford", "pelesh", "ryandanehy", "kswirydo")

```

**Review by pelesh — APPROVED** [review]
Looks good to me. I'd just add @kswirydo as a package maintainer, as well.

**Comment by kswirydo** [comment]
Looks good to me (but I have zero experience with Spack except I failed a couple times trying to use it :D )

**Review by pelesh — APPROVED** [review]
lgtm

File: var/spack/repos/builtin/packages/resolve/package.py
Comment (cameronrutherford) [reply var/spack/repos/builtin/packages/resolve/package.py]: Now using GitHub URL

**Comment by cameronrutherford** [comment]
Good to go with new public GitHub URL

**Comment by cameronrutherford** [comment]
@pelesh and @kswirydo please re-review. I screwed up and force pushed to my fork and deleted this, so even more motivation to merge this ASAP and iterate from here

File: var/spack/repos/builtin/packages/resolve/package.py
Comment (cameronrutherford) [inline var/spack/repos/builtin/packages/resolve/package.py:50]: Note we don't set these when `+cuda` as default values in CMake for these are both `True`

**Comment by cameronrutherford** [comment]
Good to merge

**Review by pelesh — APPROVED** [review]
Straightforward changes, looks good.

**Review by pelesh — APPROVED** [review]
lgtm

**Comment by cameronrutherford** [comment]
@haampie can this get merged? We are yet to have a released version, so is it ok to not have any reference versions for the moment?

**Comment by cameronrutherford** [comment]
Closing in favor of #40871

**Pull request closed without merging**
